examples/bioc/lib.rs

```diff
@@ -0,0 +1,157 @@
+//! # Basic Input Output Contract (BIOC)
+//!
+//! This is a Basic Input Output Contract (BIOC), which is intented to
+//! demonstrate rich i\o posibilties of ink! contracts, namely:
+//!
+//!   1. Use complicated nested input and ouput types.
+//!      This is done through the real use case example of data structure
+//!      needed to store a candle auction data.
+//!   2. Make contract fail with `ContractTrapped`.
+//!   3. Make contract fail with returning an Error.
+//!   4. Perform debug printing from contract into node's log.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use ink_lang as ink;
+
+#[ink::contract]
+mod bioc {
+    use ink_prelude::{
+        string::String,
+        vec::Vec,
+    };
+
+    /// No storage is needed for this contract.
+    #[ink(storage)]
+    #[derive(Default)]
+    pub struct Bioc {}
```

Comment:
We should have storage, too. UIs should add a feature to inspect storage, eventually. So this is also relevant to them.